### PR TITLE
replace mrclay/minify with JSqueeze (#13052)

### PR DESCRIPTION
**Pull request by AdamWill**
mrclay/minify is present only to provide the 'JSMin' minifier, but this is non-free. JSqueeze is free, it's also a currently maintained project following good development practices, and not tied to a bundle of other junk like mrclay/minify. This commit goes along with a commit to owncloud itself to switch from using Assetic's JSMin filter to a backport of its latest JSqueeze filter.

**NOTE**: this is an alternative to https://github.com/owncloud/3rdparty/pull/149 , only one or the other can sensibly be used.

The matching core PR is https://github.com/owncloud/core/pull/13229.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
tests folder is in gitignore - thx - but still checked in - can you please adjust this? THX

**Comment by AdamWill** [comment]
oh, I thought you actually wanted it that way (there is actually significance to having files in the github repo but listed in a checked-in `.gitignore` file, AIUI - they don't show up in auto-generated tarballs). I can take them out entirely if that's what you wanted, though.

**Comment by AdamWill** [comment]
Updated. For the record, here is exactly what I did:

    ./composer.phar remove --no-update mrclay/minify
    git rm -rf mrclay/
    ./composer.phar require --no-update patchwork/jsqueeze
    ./composer.phar update patchwork/jsqueeze
    git add patchwork/jsqueeze/
    git rm -rf patchwork/jsqueeze/tests/
    nano .gitignore (manual changes)

I don't know what's up with it changing the URLs to various things in `composer.lock`. composer moves in mysterious ways.

File: patchwork/jsqueeze/phpunit.xml.dist [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (MorrisJobke) [inline patchwork/jsqueeze/phpunit.xml.dist]: @AdamWill Can you add this to the .gitignore too?

**Comment by MorrisJobke** [comment]
@AdamWill Beside my comment this look good :)

**Comment by AdamWill** [comment]
Re-diffed to current with @MorrisJobke 's comment addressed. The core PR still seems to be merge-able as is.

File: composer/installed.json [hunk 2]
```diff
@@ -437,7 +437,7 @@
         },
         "dist": {
             "type": "zip",
-            "url": "https://api.github.com/repos/JamesHeinrich/getID3/zipball/178e55bc05bc78c764ae0a096c4caacf118ab12a",
+            "url": "https://api.github.com/repos/JamesHeinrich/getID3/zipball/824825682a3229db8d5f1bd8c2c0cb6aae7c1ec2",
```
Comment (LukasReschke) [inline composer/installed.json]: Looks fishy?

Comment (AdamWill) [inline composer/installed.json]: Like I said in the 2015-05-05 comment: "I don't know what's up with it changing the URLs to various things in composer.lock. composer moves in mysterious ways."

I think it's something like, even with `--no-update` on the `remove` and `require` commands, it finds out that there are newer builds of some things and updates the `.lock` file, though `--no-update` stops it actually downloading the newer builds.

If you know a better way to add something without causing changes, by all means, let me know. Again all I did was run the commands listed in that earlier comment, you can reproduce the commit simply by doing that.

File: patchwork/jsqueeze/phpunit.xml.dist (hunk 1 above)
Comment (MorrisJobke) [inline patchwork/jsqueeze/phpunit.xml.dist]: And of course also remove the file ;)

**Comment by AdamWill** [comment]
Here's some information on why those URLs change:

https://stackoverflow.com/questions/30684712/composer-lock-changes-despite-nothing-to-install-or-update

That, and the changes to the Composer classloader, come from composer itself; I run the composer commands with an updated `composer.phar`. Presumably we could avoid such changes if I used whatever version of composer other people touching `3rdparty` use, but I can't find any doc indicating that anywhere. Other commits to 3rdparty that I can see include similar 'unrelated' changes ((e.g. the recent one to phpzipstreamer).

I have rebased to current master.

**Comment by DeepDiver1975** [comment]
superseded by https://github.com/owncloud/3rdparty/pull/199

THX @AdamWill

**Pull request closed without merging**